pkg/gofr/migration/mongo.go

```diff
@@ -0,0 +1,100 @@
+package migration
+
+import (
+	"context"
+	"time"
+
+	"gofr.dev/pkg/gofr/container"
+)
+
+type mongoDS struct {
+	container.Mongo
+}
+
+type mongoMigrator struct {
+	container.Mongo
+	migrator
+}
+
+// apply initializes mongoMigrator using the Mongo interface.
+func (ds mongoDS) apply(m migrator) migrator {
+	return mongoMigrator{
+		Mongo:    ds.Mongo,
+		migrator: m,
+	}
+}
+
+const (
+	mongoMigrationCollection = "gofr_migrations"
+)
+
+// checkAndCreateMigrationTable initializes a MongoDB collection if it doesn't exist.
+func (mg mongoMigrator) checkAndCreateMigrationTable(_ *container.Container) error {
+	err := mg.Mongo.CreateCollection(context.Background(), mongoMigrationCollection)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getLastMigration retrieves the latest migration version from MongoDB.
+func (mg mongoMigrator) getLastMigration(c *container.Container) int64 {
+	var lastMigration int64
+
+	var migrations []struct {
+		Version int64 `bson:"version"`
+	}
+
+	filter := map[string]interface{}{}
```

Comment:
This would be easier to read

```suggestion
	var filter map[string]interface{}
```

Or

```suggestion
	var filter map[string]any
```

But if you want to have a non-nil slice, this would be more idiomatic

```suggestion
	filter := make(map[string]any)
```